File: in_Java/0844 Backspace String Compare.java
class Solution {
    public boolean backspaceCompare(String S, String T) {
        Stack<Character> st1 = new Stack<>();
        Stack<Character> st2 = new Stack<>();
        for(char c : S.toCharArray()){
            if(c == '#'){
                if(!st1.isEmpty()) st1.pop();
            }else{
                st1.push(c);
            }
        }
        for(char c : T.toCharArray()){
            if(c == '#'){
                if(!st2.isEmpty()) st2.pop();
            }else{
                st2.push(c);
            }
        }
        if(st1.size() != st2.size()) return false;
        while(!st1.isEmpty()){
            if(st1.pop() != st2.pop()) return false;
        }
        return true;
    }
}

File: in_Python/0844 Backspace String Compare.py
class Solution(object):
    def backspaceCompare(self, S, T):
        """
        :type S: str
        :type T: str
        :rtype: bool
        """
        s1 = []
        s2 = []
        for i in S:
            if i == '#':
                if len(s1) != 0:
                    s1.pop()
            else:
                s1.append(i)
        for i in T:
            if i == '#':
                if len(s2) != 0:
                    s2.pop()
            else:
                s2.append(i)            
        if len(s1) != len(s2):
            return False
        while len(s1) != 0:
            if s1.pop() != s2.pop():
                return False
        return True